### PR TITLE
Build: increase the amount of time we wait for a node to startup

### PR DESCRIPTION
**Pull request by jaymode**
This commit increases the amount of time we wait for a node to startup to 45 seconds from 30
seconds. This is needed as some QA tests with plugins and tribe timeout when starting the tribe
node. Even regular node startup time with plugins is starting to approach the limit of 30 seconds.

Additionally, this commit provides better feedback when a wait has failed. The code checks to
ensure all expected files exist for each node; if they do not then we consider the wait task as
having failed. Prior to this change, when there was one or more missing file the build would
continue and attempt to execute the wait condition that typically makes an HTTP request to the
cluster. The output of this type of failure does include which files exist and which do not but
this change makes it clearer that the actual HTTP call did not time out, but the failure was before
the call was even made.

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy [hunk 1]
```diff
@@ -610,6 +610,17 @@ class ClusterFormationTasks {
...
             }
 
+            // make sure all files exist otherwise we haven't fully started up
+            boolean missingFile = false
+            for (NodeInfo node : nodes) {
+                missingFile |= node.pidFile.exists() == false
+                missingFile |= node.httpPortsFile.exists() == false
+                missingFile |= node.transportPortsFile.exists() == false
+            }
+            if (missingFile) {
```
Comment (jasontedor) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: What do you think of making this a set that we remove from and spill if it's not empty at the end? Then we can give indication of what is missing?

**Review by jasontedor — COMMENTED** [review]
I left a suggestion.

**Review by nik9000 — COMMENTED** [review]
I really like checking for the files. I'm unsure about bumping the time for all tests. Maybe we should make it configurable and bump it on the tests that need it? I mean, it isn't much harm to wait the extra 15 seconds in the rare case that the node isn't starting up but I think we want to know if a test *expects* to be slow. So the act of configuring something beyond the 30 second limit communicates that we're ok with how long the test cluster takes to start.

Or not. I'm fine with 45 seconds too. I expect 30 seconds was pretty arbitrary in the first place.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy (hunk 1 above)
Comment (jaymode) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: I'm good with that but to be clear we will be duplicating data that is already printed out. `waitFailed` prints the following lines:

```java
logger.error("|  failure marker exists: ${node.failedMarker.exists()}")
logger.error("|  pid file exists: ${node.pidFile.exists()}")
logger.error("|  http ports file exists: ${node.httpPortsFile.exists()}")
logger.error("|  transport ports file exists: ${node.transportPortsFile.exists()}")
```

Comment (jasontedor) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: Oh, that's fine then.

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**